provdebug/Serializer.py

```diff
@@ -0,0 +1,27 @@
+import platform
+import json
+import random
+from .DebugRecord import DebugRecord
+from typing import List, Dict
+
+class Serializer:
+
+    @staticmethod
+    def save_records(records: List[DebugRecord]):
+        record_dicts = [r.writeDict() for r in records]
+        metadata = Serializer.platform_metadata()
+        replay_dict = {} 
+        replay_dict["metadata"] = metadata
+        replay_dict["records"] = record_dicts
+        replay_file = json.dumps(replay_dict)
+        f = open(f"./debugging-trace-{random.randint(0, 9)}.replay", "w")
+        f.write(replay_file)
+        f.close()
+
+    @staticmethod
+    def platform_metadata() -> Dict[str, str]:
+        metadata = {}
+        metadata["os"] = platform.platform()
+        metadata["python-version"] = platform.python_version()
```

Comment:
This is the metadata we collect for now. 

I'm not sure if I can get the version of R that's installed but I'll look into it. Can you think of any other metadata that might be useful?

Reply:
Maybe the author of the debugging trace?
Also, I just tried looking it up and am not sure about finding the R version either. My first thought was that it would be included in the PROV-JSON file but it looks like it's not. I guess we could consider modifying rdt to produce an extended PROV-JSON that includes the R version? But then it would need to know the version of R used to execute that original R script too. 

Also, should we change this so that it only displays the Python version if the debugging trace was created from debugging a Python script / the extended PROV-JSON is provenance collected from a Python script?

Reply:
> Maybe the author of the debugging trace?

Thinking of our use case, wouldn't the user of the trace already know who executed it?

>  I guess we could consider modifying rdt to produce an extended PROV-JSON that includes the R version? But then it would need to know the version of R used to execute that original R script too.

Time to learn some R lmao

> Also, should we change this so that it only displays the Python version if the debugging trace was created from debugging a Python script / the extended PROV-JSON is provenance collected from a Python script?

Yep, you're definitely right

Reply:
> Thinking of our use case, wouldn't the user of the trace already know who executed it?
That's true. I think I was thinking about that along the lines of the issue report generation feature that might come along later but that doesn't need to be included in the trace itself.